src/Access/Authentication.h

```diff
@@ -103,7 +104,11 @@ class Authentication
 
     Type type = Type::NO_PASSWORD;
     Digest password_hash;
+
+    // Used and maintained only for LDAP.
     String server_name;
+    mutable std::size_t last_successful_password_check_params_hash = 0;
+    mutable std::chrono::steady_clock::time_point last_successful_password_check_timestamp;
```

Comment:
I don't like that. The `Authentication` class shouldn't contain any temporary information. It's designed to be a simple structure which should be copied and compared easily. Why don't move these two variables to `ExternalAuthenticators`, probably in form of some LRUCache?

Reply:
Moving it to other place (`ExternalAuthenticators` or other) will raise synchronization, state maintenance, and lookup issues and will complicate things way more than necessary. `Authentication` instances are still easily copied, and, conceptually, the info that is stored here is **equivalent** to the password, that could be stored in clear text.

Reply:
Pushed a fix.